qiita_pet/handlers/study_handlers/sample_template.py

```diff
@@ -203,11 +206,16 @@ def sample_template_handler_patch_request(user, req_op, req_path,
                 raise HTTPError(400, reason="Value is required when updating "
                                 "sample information")
 
-            # Check if the file exists
-            fp_rsp = check_fp(study_id, req_value)
-            if fp_rsp['status'] != 'success':
-                raise HTTPError(404, reason='Filepath not found')
-            filepath = fp_rsp['file']
+            if direct_upload:
+                # We can assume that the file exist as it was generated by
```

Comment:
I don't believe this is safe. If the file is going onto a NFS mount, then just because it's uploaded does not mean all hosts mounting the share can see the file.

Reply:
Also, this method isn't implied to be private or encapsulated, so this is really more more in hope that the caller has taken care of this...

Reply:
Good point, didn't consider originally. However, in Qiita this is safe as the direct-upload folder uses mkstemp, which uses TMPDIR, which is shared in the system.

Reply:
Is `TMPDIR` assured to be a local filesystem?

Reply:
No, it's a mount folder, just like the upload folder for any study. I can add an explicit check (if exists) and raise an error but it will look pretty similar to the error the regular open will raise.

Reply:
may be nice to be explicit, and to catch a pathological situation as early as possible though?

Reply:
Sure ... will add an error saying to contact qiita.help immediately ...